utils/srcmanip/check_import_usage

```diff
@@ -0,0 +1,110 @@
+#!/usr/bin/env python3
+
+"""Simple script to ckeck usage of imports in Fortran files"""
+
+import argparse
+import fnmatch
+import os
+import re
+
+
+_DESCRIPTION = "Checks the imports in DFTB+ Fortran file(s) for usage"
+
+_PAT_USE_MODULE = re.compile(
+    r"""^(?P<indent>[ ]*)use
+    (?P<attrib>(?:\s*,\s*intrinsic)?)
+    (?P<separator>[ ]*::[ ]*|[ ]*)
+    (?P<name>\w+)
+    (?P<rest>.*?(?:&[ ]*\n(?:[ ]*&)?.*?)*)\n
+    """,
+    re.VERBOSE | re.MULTILINE,
+)
+
+WORD_PATTERN = re.compile(r"""\w+""")
+
+
+def main():
+    """Main script driver."""
+    args = parse_arguments()
+    filenames = []
+    if args.folders:
+        filenames = get_files(args.folders)
+    if args.files:
+        filenames += args.files
+    for fname in filenames:
+        with open(fname, "r", encoding="utf-8") as file:
+            file_txt = file.read()
+            matches = list(_PAT_USE_MODULE.finditer(file_txt))
+            matches.reverse()
+            rest_list = []
+            for match in matches:
+                if match["rest"]:
+                    rest_list.append(match["rest"])
+                file_txt = file_txt[: match.start()] + file_txt[match.end() :]
```

Comment:
What's the point to check the source *before* the import? It is enough to check only the text which comes after the match.

Reply:
Could, you, please also try to rebase it to current main?